pyodide-build/pyodide_build/pywasmcross.py

```diff
@@ -136,24 +141,25 @@ def capture_make_command_wrapper_symlinks(env: Dict[str, str]):
         env[var] = symlink
 
 
-def capture_compile(args: argparse.Namespace):
+def capture_compile(*, host_install_dir: str, skip_host: bool, env: Dict[str, str]):
     TOOLSDIR = Path(common.get_make_flag("TOOLSDIR"))
-    env = dict(os.environ)
-    capture_make_command_wrapper_symlinks(env)
+    env["PYODIDE"] = "1"
```

Comment:
I think it might be better to make a copy of the env object before making any changes to it, similarly to what was done before?

Reply:
Here we are getting `bash_runner.env`, so changing the environment in place means it applies to the post script. I think it would be a good idea to move the `PYODIDE=1` out into `get_bash_runner` in buildpkg. The path modification should indeed happen to a copied version -- if people invoke the compiler in the `post` script, it would be an unwelcome surprise to be invoking the fake compiler.